src/index.ts

```diff
@@ -1,73 +1,102 @@
+  },
+  // "Polling running services is dangerous and not recommended in production.
+  // Polling should only be used against a registry.
+  // If you are polling running services, use with caution"
+  // - https://github.com/apollographql/apollo-server/discussions/4280.
+  // experimental_pollInterval: 600000, // every 10 min
 });
 
-
 (async () => {
-    const server = new ApolloServer({
-        gateway,
-        subscriptions: false,
-        context: ({ req }) => {
-            const apiTokens: string = req.headers["api-tokens"] || "";
-            const acceptLanguage: string = req.headers["accept-language"] || "";
-            return { apiTokens, acceptLanguage };
-        },
-        debug: debug,
-        playground: debug,
-        introspection: debug
-    });
+  const server = new ApolloServer({
+    gateway,
+    subscriptions: false,
+    context: ({ req }) => {
+      const apiTokens: string = req.headers["api-tokens"] || "";
+      const acceptLanguage: string = req.headers["accept-language"] || "";
+      return { apiTokens, acceptLanguage };
+    },
+    debug: debug,
+    playground: debug,
+    introspection: debug,
+  });
 
-    const app = express();
+  const app = express();
 
-    app.use(cors());
+  app.use(cors());
 
-    // GraphQL Voyager schema visualization
-    if (debug) {
-        const voyagerMiddleware = require('graphql-voyager/middleware').express;
-        app.use('/voyager', voyagerMiddleware({
-            endpointUrl: '/graphql',
-            displayOptions: {
-                sortByAlphabet: true,
-            },
-        }));
-    }
+  // GraphQL Voyager schema visualization
+  if (debug) {
+    const voyagerMiddleware = require("graphql-voyager/middleware").express;
+    app.use(
+      "/voyager",
+      voyagerMiddleware({
+        endpointUrl: "/graphql",
+        displayOptions: {
+          sortByAlphabet: true,
+        },
+      })
+    );
+  }
+
+  // TODO: check that app actually works
+  app.get("/readiness", (req, res) => {
+    res.status(200).json({ status: "OK" });
+  });
 
-    // TODO: check that app actually works
-    app.get("/readiness", ( req, res ) => {
-        res.status(200).json({status: "OK"});
-    });
+  app.get("/healthz", async (req, res) => {
+    const isBerthApiHealthy = await testConnectionToBerthReservationsBackend();
+    const isProfileApiHealthy = await testConnectionToOpenCityProfileBackend();
 
-    app.get( "/healthz", ( req, res ) => {
-        res.status(200).json({status: "OK"});
-    });
+    if (!isBerthApiHealthy || !isProfileApiHealthy) {
+      let messages: string[] = [];
+      if (!isBerthApiHealthy) {
+        messages.push("Connection issues with the Berth API.");
+      }
+      if (!isProfileApiHealthy) {
+        messages.push("Connection issues with the Open City Profile API.");
+      }
+      // 504 Gateway Timeout
+      res.status(504).json({ status: "ERROR", messages });
+    } else {
+      res.status(200).json({ status: "OK" });
+    }
```

Comment:
Minor. In case it's important to you, there's an opportunity to have less nesting here.

```ts
    let messages: string[] = [];
    if (!isBerthApiHealthy) {
      messages.push("Connection issues with the Berth API.");
    }
    if (!isProfileApiHealthy) {
      messages.push("Connection issues with the Open City Profile API.");
    }

    // 504 Gateway Timeout
    if (message.length > 0) {
      res.status(504).json({ status: "ERROR", messages });
    } else {
      res.status(200).json({ status: "OK" });
    }
```